### PR TITLE
Added Kubernetes validation for Tinkerbell hardware name field

### PR DESCRIPTION
**Pull request by panktishah26**
*Description of changes:*
Added Kubernetes validation for Tinkerbell hardware name field for the generate hardware and create cluster command.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cmd/eksctl-anywhere/cmd/generatehardware.go [hunk 1]
```diff
@@ -132,6 +132,13 @@ func (hOpts *hardwareOptions) generateHardware(ctx context.Context) error {
 			return fmt.Errorf("failed reading csv: %v", err)
 		}
 
+		if !hOpts.dryRun {
+			err := validation.IsDNS1123Subdomain(items[csv.HostnameIndex])
+			if err != nil {
+				return fmt.Errorf("error validating hardware hostname, hostname %s is invalid %v", items[csv.HostnameIndex], err)
```
Comment (mitalipaygude) [inline cmd/eksctl-anywhere/cmd/generatehardware.go:138]: We don't really need the keyword `error` while already returning an `Errorf`

File: pkg/providers/tinkerbell/hardware/validator.go [hunk 2]
```diff
@@ -93,6 +97,11 @@ func UniqueHostnames() MachineAssertion {
 			return fmt.Errorf("duplicate Hostname: %v", m.Hostname)
 		}
 
+		err := apimachineryvalidation.IsDNS1123Subdomain(m.Hostname)
```
Comment (chrisdoherty4) [inline pkg/providers/tinkerbell/hardware/validator.go:100]: This check is already being done on the `Machine`s `Validate` method. I think the additional check you added was when we read in the yamls again - I've not touched that yet. 

Also, side note, this function is self contained and single responsibility. Ideally we shouldn't add to these instead adding new validation functions.

Comment (panktishah26) [reply pkg/providers/tinkerbell/hardware/validator.go:100]: I used to think that you have added validation for hardware yaml file and not the generate hardware file. I have corrected the code.

**Review by chrisdoherty4 — APPROVED** [review]

**Pull request merged**